src/main/java/org/fogbowcloud/manager/core/threads/ClosedComputeOrdersProcessor.java

```diff
@@ -0,0 +1,110 @@
+package org.fogbowcloud.manager.core.threads;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
+import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
+import org.fogbowcloud.manager.core.datastructures.SharedOrderHolders;
+import org.fogbowcloud.manager.core.exceptions.OrderStateTransitionException;
+import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
+import org.fogbowcloud.manager.core.models.linkedList.ChainedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+
+import java.util.Map;
+import java.util.Properties;
+
+public class ClosedComputeOrdersProcessor implements Runnable {
+
+    private static final Logger LOGGER = Logger.getLogger(ClosedComputeOrdersProcessor.class);
+
+    private ChainedList closedOrders;
+
+    private String localMemberId;
+    private Long sleepTime;
+
+    private InstanceProvider localProvider;
+    private InstanceProvider remoteProvider;
+
+    private SharedOrderHolders sharedOrderHolders = SharedOrderHolders.getInstance();
+
+    public ClosedComputeOrdersProcessor(Properties properties,InstanceProvider localProvider,
+                                        InstanceProvider remoteProvider) {
+        SharedOrderHolders sharedOrdersHolder = SharedOrderHolders.getInstance();
+        this.closedOrders = sharedOrdersHolder.getClosedOrdersList();
+
+        String closedOrdersSleepTime = properties.getProperty(ConfigurationConstants.CLOSED_ORDERS_SLEEP_TIME_KEY,
+                DefaultConfigurationConstants.CLOSED_ORDERS_SLEEP_TIME);
+        this.sleepTime = Long.valueOf(closedOrdersSleepTime);
+
+        this.localMemberId = properties.getProperty(ConfigurationConstants.XMPP_ID_KEY);
+
+        this.localProvider = localProvider;
+        this.remoteProvider = remoteProvider;
+    }
+
+    @Override
+    public void run() {
+        boolean isActive = true;
+        while (isActive) {
+            try {
+                Order order = this.closedOrders.getNext();
+                if (order != null) {
+                    try {
+                        processClosedOrder(order);
+                        closedOrders.removeItem(order);
```

Comment:
Why you are removing the order from closed list in the run method?

Reply:
Please take a look at the last page of our Design Contract. The pseudo-code of this member is detailed there.

Reply:
Sorry, you do not understand my question, I know that is necessary to remove the order from the closed orders list, I am questioning why you are doing it at the run method? Is more appropriated to do it at the process method?